src/Controller/EnMarche/CommitteeController.php

```diff
@@ -169,7 +172,68 @@ public function unfollowAction(Request $request, Committee $committee): Response
             'button' => [
                 'label' => 'Suivre ce comité',
                 'action' => 'rejoindre',
-                'csrf_token' => (string) $this->get('security.csrf.token_manager')->getToken('committee.follow'),
+                'csrf_token' => (string) $this->get('security.csrf.token_manager')->getToken('committee.unfollow'),
+            ],
+        ]);
+    }
+
+    /**
+     * @Route("/voter", name="app_committee_vote", condition="request.request.has('token')", methods={"POST"})
+     * @Security("is_granted('VOTE_COMMITTEE', committee)")
+     */
+    public function voteAction(
+        Request $request,
+        Committee $committee,
+        ValidatorInterface $validator,
+        CsrfTokenManager $tokenManager,
+        SerializerInterface $serializer
+    ): Response {
+        if (!$this->isCsrfTokenValid('committee.vote', $request->request->get('token'))) {
+            throw $this->createAccessDeniedException('Invalid CSRF protection token to vote committee.');
+        }
+
+        $membership = $this->getCommitteeManager()->enableVoteInCommittee($this->getUser(), $committee);
+        $errors = $validator->validate($membership);
+
+        if (0 !== \count($errors)) {
+            $jsonErrors = $serializer->serialize($errors, 'jsonproblem');
+
+            return JsonResponse::fromJsonString($jsonErrors, Response::HTTP_BAD_REQUEST);
+        }
+
+        $this->getDoctrine()->getManager()->flush();
```

Comment:
Could you inject all needed services and avoid to call `Controller::getService`?